client/extensions/wp-job-manager/components/setup/index.jsx

```diff
@@ -18,40 +19,49 @@ import PageSetup from './page-setup';
 import Wizard from 'components/wizard';
 import { getSelectedSiteSlug } from 'state/ui/selectors';
 
-const SetupWizard = ( {
-	slug,
-	stepName = Steps.INTRO,
-	translate,
-} ) => {
-	const steps = [ Steps.INTRO, Steps.PAGE_SETUP, Steps.CONFIRMATION ];
-	const components = {
-		[ Steps.INTRO ]: <Intro />,
-		[ Steps.PAGE_SETUP ]: <PageSetup />,
-		[ Steps.CONFIRMATION ]: <Confirmation />,
+class SetupWizard extends Component {
+	static propTypes = {
+		slug: PropTypes.string,
+		stepName: PropTypes.string,
+		translate: PropTypes.func.isRequired,
 	};
-	const mainClassName = 'wp-job-manager__setup';
 
-	return (
-		<Main className={ mainClassName }>
-			<DocumentHead title={ translate( 'Setup' ) } />
-			<Wizard
-				basePath={ `/extensions/wp-job-manager/setup/${ slug }` }
-				components={ components }
-				forwardText={ translate( 'Continue' ) }
-				steps={ steps }
-				stepName={ stepName } />
-		</Main>
-	);
-};
+	goToStep = ( event, step ) => {
+		event.preventDefault();
+		page( `/extensions/wp-job-manager/setup/${ this.props.slug }/${ step }` );
+	}
+
+	render() {
+		const steps = [ Steps.INTRO, Steps.PAGE_SETUP, Steps.CONFIRMATION ];
+		const components = {
+			[ Steps.INTRO ]: <Intro goToStep={ this.goToStep } />,
+			[ Steps.PAGE_SETUP ]: <PageSetup goToStep={ this.goToStep } />,
+			[ Steps.CONFIRMATION ]: <Confirmation />,
+		};
+		const {
+			slug,
+			stepName = Steps.INTRO,
+			translate,
+		} = this.props;
+		const mainClassName = 'wp-job-manager__setup';
+
+		return (
+			<Main className={ mainClassName }>
+				<DocumentHead title={ translate( 'Setup' ) } />
+				<Wizard
+					basePath={ `/extensions/wp-job-manager/setup/${ slug }` }
```

Comment:
I wonder if we could avoid this hardcoded path somehow? Or at least extract it so it isn't duplicated between this line and line 31. Maybe a helper method?

```javascript
function wizardComponentPath( slug, step = '' ) {
  return `/extensions/wp-job-manager/setup/${ this.props.slug }/${ step }`;
}
```

Reply:
Using `settingsPath` from the extension's config seems like the way to go here.
[Here](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/zoninator/app/util.js#L18) is an example of how a helper method for that could look like.  
It would be cool to address that in a separate PR and update any similar spots in the extension as well.

Reply:
I've put it in a `const` for now - https://github.com/Automattic/wp-calypso/pull/18128/commits/c284810501e89baea1b1bf2ce1b3c4ef2ac87f8f#diff-13b72cd7cc1349f2fd2cda5eda4d3089R7